### PR TITLE
Support Serial2 

### PR DESCRIPTION
**Pull request by ogatatsu**

### CODE REVIEW

**Comment by hathach** [comment]
thanks for the PR @ogatatsu . Look like you change the Serial from UART to UARTE, though the travis  build is failed https://travis-ci.com/adafruit/Adafruit_nRF52_Arduino/builds/121871228
Since Serial affect pretty much all the usage of this repo. Would you mind fixing the travis build, I will manually test it more once travis is fixed.

**Comment by ogatatsu** [comment]
I am sorry, I did not check the environment other than PCA10056.

**Comment by elral** [comment]
I can confirm that these changes on an Adafruit nRF52840 Feather also works after modifying the variant.h accordingly.

**Comment by joseangeljimenez** [comment]
@hathach @ogatatsu Is there anything I can help with in order to speed up the merging of this pull request? I am available for testing.

**Comment by hathach** [comment]
Thank you very much, If you could verify it work with various scenarios and several of stock sketches especially when hooking with gps breakout/wing. That would be good enough. I was too busy with usb work previously and kind of forget to test this issue out (my bads). I am currently on vacation, I will try this out and merge this asap after some minimal testing

File: cores/nRF5/Uart.cpp [hunk 1]
```diff
@@ -55,29 +55,29 @@ Uart::Uart(NRF_UART_Type *_nrfUart, IRQn_Type _IRQn, uint8_t _pinRX, uint8_t _pi
 
 void Uart::setPins(uint8_t pin_rx, uint8_t pin_tx)
 {
-  uc_pinRX = pin_rx;
-  uc_pinTX = pin_tx;
+  uc_pinRX = g_ADigitalPinMap[pin_rx];
```
Comment (hathach) [inline cores/nRF5/Uart.cpp]: nice catch :)

File: cores/nRF5/Uart.cpp [hunk 2]
```diff
@@ -162,15 +163,28 @@ void Uart::end()
 
 void Uart::flush()
 {
-	rxBuffer.clear();
+  if ( _begun ) {
+    xSemaphoreTake(_mutex, portMAX_DELAY);
+    xSemaphoreGive(_mutex);
```
Comment (hathach) [inline cores/nRF5/Uart.cpp]: buffer clear is probably missing :)

**Pull request merged**